### DISCUSSION TITLE
Always on

### DISCUSSION THREAD

**Discussion by Tiflit** [7gxycn08/PyAutoActions#6] [Q&A] [answered]
Hi,
Is there a way to keep HDR on at all times instead of shutting it down after opening the tool/exiting an application?
It could be added as an option in the menus (if you take suggestions). I have no idea how to do it myself. :/

In any case I want to thank you immensely for this work. HDR being difficult to use on Windows, games and emulators now work perfectly with PyAutoActions.

**Comment by 7gxycn08** [comment]
Glad you find the tool useful :) 
Can you explain the always on feature in detail and what problem would it solve if Implemented?

Like what you mean by always on example: Does the program sit there and check hdr status if it's off it will turn it on.

**Reply by Tiflit** [reply]
Exactly. When PyAuto starts:
If HDR is already on; do nothing.
If HDR is off; turn it on.
Repeat when launching an app.

I suggested adding it as an optional user option because the Readme says it is supposed to also turn off HDR when exiting an app.

When PyAuto opens up, it automatically shuts down HDR (this is what I would like it to stop doing). Then if HDR is reactivated manually with Windows settings or Win+Alt+B, it stays on no matter what afterwards. I suppose Windows supersedes PyAuto, unless something changes in the display environment (computer goes to sleep, displays are switched/removed/added, graphics settings modified, etc.) but this is not an issue.

**Reply by 7gxycn08** [reply]
> Exactly. When PyAuto starts: If HDR is already on; do nothing. If HDR is off; turn it on. Repeat when launching an app.
> 
> I suggested adding it as an optional user option because the Readme says it is supposed to also turn off HDR when exiting an app.
> 
> When PyAuto opens up, it automatically shuts down HDR (this is what I would like it to stop doing). Then if HDR is reactivated manually with Windows settings or Win+Alt+B, it stays on no matter what afterwards. I suppose Windows supersedes PyAuto, unless something changes in the display environment (computer goes to sleep, displays are switched/removed/added, graphics settings modified, etc.) but this is not an issue.

I see that global windows hdr settings is switching to off when PyAutoActions is launched due to Toggle Mode option selected. I will fix this issue by adding a toggle option in the Toggle Mode menu called Respect Global Settings.

Am i understanding the the issue correctly? 

Please check this debug version If it's working as intended.
[PyAutoActions-Debug.zip](https://github.com/user-attachments/files/17894688/PyAutoActions-Debug.zip)

**Reply by Tiflit** [reply]
It works! Thanks.

**Reply by 7gxycn08** [reply]
Will release the new option in

> It works! Thanks.

Will release the new option in v1.2.3 

should be uploaded soon.

**Reply by 7gxycn08** [reply, marked as answer]
> It works! Thanks.

New release added you can find it in releases page:
https://github.com/7gxycn08/PyAutoActions/releases/tag/v1.2.3

Please star the repository It helps and don't forget to spread the word about PyAutoActions.
Thanks.
